sphinxval/utils/duplicates.py

```diff
@@ -0,0 +1,417 @@
+    dict["CME Latitude"].append(cme_lat)
+    dict["CME Longitude"].append(cme_lon)
+    dict["CME Speed"].append(cme_speed)
+    dict["CME Half Width"].append(cme_half_width)
+    dict["CME PA"].append(cme_pa)
+    dict["CME Catalog"].append(cme_catalog)
+    dict["Number of Flares"].append(nfl)
+    dict["Flare Latitude"].append(fl_lat)
+    dict["Flare Longitude"].append(fl_lon)
+    dict["Flare Start Time"].append(fl_start_time)
+    dict["Flare Peak Time"].append(fl_peak_time)
+    dict["Flare End Time"].append(fl_end_time)
+    dict["Flare Last Data Time"].append(fl_last_data_time)
+    dict["Flare Intensity"].append(fl_intensity)
+    dict["Flare Integrated Intensity"].append(fl_integrated_intensity)
+    dict["Flare NOAA AR"].append(fl_AR)
+   
+
+    #PREDICTION INFORMATION
+    dict["Predicted SEP All Clear"].append(prediction.all_clear.all_clear_boolean)
+    dict["Predicted SEP Probability"].append(pred_prob)
+    dict["Predicted SEP Threshold Crossing Time"].append(pred_thresh_cross)
+    dict["Predicted SEP Start Time"].append(pred_start_time)
+    dict["Predicted SEP End Time"].append(pred_end_time)
+    dict["Predicted Point Intensity"].append(pred_point_intensity)
+    dict["Predicted Point Intensity Units"].append(pred_pti_units)
+    dict["Predicted Point Intensity Time"].append(pred_pti_time)
+    dict["Predicted SEP Peak Intensity (Onset Peak)"].append(pred_peak_intensity)
+    dict["Predicted SEP Peak Intensity (Onset Peak) Units"].append(pred_pi_units)
+    dict["Predicted SEP Peak Intensity (Onset Peak) Time"].append(pred_pi_time)
+    dict["Predicted SEP Peak Intensity Max (Max Flux)"].append(pred_peak_intensity_max)
+    dict["Predicted SEP Peak Intensity Max (Max Flux) Units"].append(pred_pimax_units)
+    dict["Predicted SEP Peak Intensity Max (Max Flux) Time"].append(pred_pimax_time)
+    dict["Predicted SEP Fluence"].append(pred_fluence)
+    dict["Predicted SEP Fluence Units"].append(pred_fl_units)
+    dict["Predicted SEP Fluence Spectrum"].append(pred_fl_spec)
+    dict["Predicted SEP Fluence Spectrum Units"].append(pred_flsp_units)
+    dict["Predicted Time Profile"].append(pred_time_profile)
+
+
+
+def identify_forecast_duplicates(df):
+    """ Check the Forecast dataframe for duplicate entries. Issue warning
+        and remove repeated forecasts, combined with observatory information.
+        
+        Forecasts will be considered duplicate if all fields in the
+        dataframe are exactly the same.
+        
+        Output:
+        
+            :df: (dataframe) with unique entries
+        
+    """
+    
+    #Sort the dataframe in time order
+    df = df.sort_values(by=["Model","Energy Channel Key", "Forecast Issue Time", "Prediction Window Start"],ascending=[True, True, True, True])
+
+
+    #Extract key rows from the df that uniquely identify a forecast
+    #Cannot use all df entries, because the hash command cannot hash lists.
```

Comment:
Think it would be shorter and more maintainable to enumerate the columns to **not** include and include all the rest.